packages/modules/backup_clouds/samba/backup_cloud.py

```diff
@@ -10,11 +12,16 @@
 
 
 def upload_backup(config: SambaBackupCloudConfiguration, backup_filename: str, backup_file: bytes) -> None:
-    conn = SMBConnection(config.smb_user, config.smb_password, 'client', 'server', use_ntlm_v2=True)
-    conn.connect(config.smb_path, 445)
-
-    with conn.openFile(backup_filename, 'w') as file:
-        file.write(backup_file)
+    conn = SMBConnection(config.smb_user, config.smb_password, os.uname()[1], config.smb_server, use_ntlm_v2=True)
+    log.info("SMB Verbindungsaufbau")
+    conn.connect(config.smb_server,139)
+
+    if config.smb_path is None:
+        log.info("Backup auf //" + config.smb_server + '/' + config.smb_share )
+        conn.storeFile(config.smb_share, backup_filename.replace(':',''), io.BytesIO(backup_file))
+    else:
+        log.info("Backup auf //" + config.smb_server + '/' + config.smb_share + '/' + config.smb_path )
+        conn.storeFile(config.smb_share, config.smb_path + backup_filename.replace(':',''), io.BytesIO(backup_file))
```

Comment:
```suggestion
    smb_filename = backup_filename.replace(':', '_')
    full_file_path = config.smb_path + smb_filename if config.smb_path is not None else smb_filename
    log.info("Backup nach //" + config.smb_server + '/' + config.smb_share + full_file_path)
    conn.storeFile(config.smb_share, full_file_path, io.BytesIO(backup_file))
```
So wird mehrfach vorhandener Code (backup_filename.replace()`) vermieden. Ebenfalls ist es üblich, nicht erlaubte Zeichen durch einen Unterstrich zu ersetzen, anstatt komplett zu entfernen.